src/main/java/ai/elimu/rest/v1/project/AppCollectionRestController.java

```diff
@@ -0,0 +1,102 @@
+import ai.elimu.model.project.AppGroup;
+import ai.elimu.model.project.License;
+import ai.elimu.rest.v1.JavaToGsonConverter;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.log4j.Logger;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Returns the {@link AppCollection} for a given {@link License}.
+ */
+@RestController
+@RequestMapping(value = "/rest/v1/project/app-collections/{appCollectionId}", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+public class AppCollectionRestController {
+    
+    private Logger logger = Logger.getLogger(getClass());
+    
+    @Autowired
+    private LicenseDao licenseDao;
+    
+    @Autowired
+    private AppCollectionDao appCollectionDao;
+    
+    @Autowired
+    private ApplicationVersionDao applicationVersionDao;
+    
+    @RequestMapping(value = "/applications", method = RequestMethod.GET)
+    public String getApplications(
+            HttpServletRequest request,
+            @PathVariable Long appCollectionId,
+            @RequestParam String licenseEmail,
+            @RequestParam String licenseNumber) {
+        logger.info("read");
+        
+        logger.info("request.getQueryString(): " + request.getQueryString());
+        logger.info("request.getRemoteAddr(): " + request.getRemoteAddr());
+        
+        JSONObject jsonObject = new JSONObject();
+        
+        License license = licenseDao.read(licenseEmail, licenseNumber);
+        if (license == null) {
+            jsonObject.put("result", "error");
+            jsonObject.put("description", "Invalid license");
+        } else {
+            AppCollection appCollection = appCollectionDao.read(appCollectionId);
+            if (appCollection == null) {
+                jsonObject.put("result", "error");
+                jsonObject.put("description", "AppCollection not found");
+            } else {
+                // TODO: verify that the AppCollection matches the one associated with the provided License
+                
+                JSONArray applications = new JSONArray();
+                
+                for (AppCategory appCategory : appCollection.getAppCategories()) {
+                    for (AppGroup appGroup : appCategory.getAppGroups()) {
+                        for (Application application : appGroup.getApplications()) {
+                            ApplicationGson applicationGson = JavaToGsonConverter.getApplicationGson(application);
```

Comment:
Line 78 to 989 is duplicate code from ApplicationRestController. Would it be better to move it to a separate ApplicationVersionHelper? Or ApplicationVersionService? Which will use dao